### PR TITLE
resource/aws_organizations_organization: Add enabled_policy_types argument

### LINKED ISSUE #4545: Organizations: Add Mechanism to enable Service control policies on the root
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Need a mechanism to enable Service control policies on the root. It looks like this needs to be done manually before you can successfully apply policies with Organizations (Organizations, Organize accounts, click the Root on the left pane, click Enable under ENABLE/DISABLE POLICY TYPES, Service control policies.

Attempting to add policies without toggling this setting results in this:
```
aws_organizations_policy_attachment.root_FullAccess: Creating...
  policy_id: "" => "p-FullAWSAccess"
  target_id: "" => "r-SECRET"
Releasing state lock. This may take a few moments...

Error: Error applying plan:

1 error(s) occurred:

* aws_organizations_policy_attachment.root_FullAccess: 1 error(s) occurred:

* aws_organizations_policy_attachment.root_FullAccess: error creating Organizations Policy Attachment: PolicyTypeNotEnabledException: This operation can be performed only for enabled policy types.
        status code: 400, request id: 50573131-5866-11e8-a4c8-2f34931e1acc
```

### References

https://docs.aws.amazon.com/organizations/latest/userguide/orgs_manage_policies.html#enable_policies_on_root

### PR DESCRIPTION
**Pull request by bflad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Closes #4545

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource/aws_organizations_organization: Add enabled_policy_types argument
```

The `aws_organizations_policy_attachment` acceptance testing was previously written to assume that the account running it was already in an Organization, manually had enabled Service Control Policies in the Root, and did not check Root or Organizational Unit policy attachments as the appropriate attributes/resources did not previously exist. The updates to those tests now verify the SCP attachment workflow end-to-end with creating a new Organization, enabling SCPs, and SCP attachments to an account, Root, and OU.

Previous output from acceptance testing (before `enabled_policy_types` implementation):

```
    --- FAIL: TestAccAWSOrganizations/PolicyAttachment (37.12s)
        --- FAIL: TestAccAWSOrganizations/PolicyAttachment/Account (14.00s)
            testing.go:568: Step 0 error: errors during apply:

                Error: error creating Organizations Policy Attachment: PolicyTypeNotEnabledException: This operation can be performed only for enabled policy types.
                  status code: 400, request id: 381509e0-7225-11e9-b974-09edfb312bea

                  on /var/folders/v0/_d108fkx1pbbg4_sh864_7740000gn/T/tf-test737240811/main.tf line 11:
                  (source code not available)

        --- FAIL: TestAccAWSOrganizations/PolicyAttachment/OrganizationalUnit (10.85s)
            testing.go:568: Step 0 error: errors during apply:

                Error: error creating Organizations Policy Attachment: PolicyTypeNotEnabledException: This operation can be performed only for enabled policy types.
                  status code: 400, request id: 3f587964-7225-11e9-96c5-1d623fb91cbf

                  on /var/folders/v0/_d108fkx1pbbg4_sh864_7740000gn/T/tf-test570985045/main.tf line 16:
                  (source code not available)

        --- FAIL: TestAccAWSOrganizations/PolicyAttachment/Root (12.27s)
            testing.go:568: Step 0 error: errors during apply:

                Error: error creating Organizations Policy Attachment: PolicyTypeNotEnabledException: This operation can be performed only for enabled policy types.
                  status code: 400, request id: 46589efd-7225-11e9-b974-09edfb312bea

                  on /var/folders/v0/_d108fkx1pbbg4_sh864_7740000gn/T/tf-test865604943/main.tf line 11:
                  (source code not available)
```

Output from acceptance testing:

```
    --- PASS: TestAccAWSOrganizations/Organization (79.29s)
        --- PASS: TestAccAWSOrganizations/Organization/basic (13.66s)
        --- PASS: TestAccAWSOrganizations/Organization/AwsServiceAccessPrincipals (24.59s)
        --- PASS: TestAccAWSOrganizations/Organization/EnabledPolicyTypes (30.29s)
        --- PASS: TestAccAWSOrganizations/Organization/FeatureSet (10.75s)
    --- PASS: TestAccAWSOrganizations/PolicyAttachment (58.58s)
        --- PASS: TestAccAWSOrganizations/PolicyAttachment/Account (21.28s)
        --- PASS: TestAccAWSOrganizations/PolicyAttachment/OrganizationalUnit (20.48s)
        --- PASS: TestAccAWSOrganizations/PolicyAttachment/Root (16.82s)
```

### CODE REVIEW

File: aws/resource_aws_organizations_organization.go [hunk 1]
```diff
@@ -221,6 +273,47 @@ func resourceAwsOrganizationsOrganizationUpdate(d *schema.ResourceData, meta int
...
+			if err := waitForOrganizationDefaultRootPolicyTypeDisable(conn, policyType); err != nil {
+				return fmt.Errorf("error waiting for policy type (%s) disabling in Organization (%s) Root (%s): %s", policyType, d.Id(), defaultRootID, err)
+			}
+		}
+
+		for _, v := range newSet.Difference(oldSet).List() {
+			policyType := v.(string)
+			input := &organizations.EnablePolicyTypeInput{
+				PolicyType: aws.String(policyType),
+				RootId:     aws.String(d.Get("roots.0.id").(string)),
```
Comment (tombuildsstuff) [inline aws/resource_aws_organizations_organization.go]: _minor_ we can use `defaultRootID` here?

File: aws/resource_aws_organizations_organization.go [hunk 2]
```diff
@@ -181,6 +221,18 @@ func resourceAwsOrganizationsOrganizationRead(d *schema.ResourceData, meta inter
 		return fmt.Errorf("error setting aws_service_access_principals: %s", err)
 	}
 
+	enabledPolicyTypes := make([]string, 0)
+
+	for _, policyType := range roots[0].PolicyTypes {
+		if aws.StringValue(policyType.Status) == organizations.PolicyTypeStatusEnabled || aws.StringValue(policyType.Status) == organizations.PolicyTypeStatusPendingEnable {
```
[the remark this answers is not in the document]
Comment (bflad) [reply aws/resource_aws_organizations_organization.go]: A lot of the AWS resources currently treat pending statuses similar to their end state during read to account for scenarios where changes might have occurred outside Terraform or for eventual consistency reasons. This one was a holdout before I implemented the waiter functions as during my local testing of this it didn't appear to display any eventual consistency issues. I'll remove it and verify everything still looks okay. 👍

File: aws/resource_aws_organizations_organization.go (hunk 1 above)
Comment (bflad) [reply aws/resource_aws_organizations_organization.go]: Yes we can! Thanks!

**Comment by bflad** [comment]
Output from acceptance testing after above minor adjustments:

```
    --- PASS: TestAccAWSOrganizations/Organization (84.63s)
        --- PASS: TestAccAWSOrganizations/Organization/EnabledPolicyTypes (34.08s)
        --- PASS: TestAccAWSOrganizations/Organization/FeatureSet (11.33s)
        --- PASS: TestAccAWSOrganizations/Organization/basic (13.29s)
        --- PASS: TestAccAWSOrganizations/Organization/AwsServiceAccessPrincipals (25.93s)
```

Merging!

**Comment by bflad** [comment]
This has been released in [version 2.10.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2100-may-10-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**